src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node heap;
+    private Node tall;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            Node<T> newNode = new Node<T>(null, value, null);
+            heap = newNode;
+            tall = newNode;
+        } else {
+            Node<T> newNode = new Node<T>(tall, value, null);
+            tall.next = newNode;
+            tall = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index was failed ");
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, heap);
+            heap.prev = newNode;
+            heap = newNode;
+            size++;
+        } else {
+            Node<T> newNode = new Node<T>(getNode(index - 1), value, getNode(index));
```

Comment:
when we call the getNode() method, we always go through the whole list to some node. It increases the complexity of this method. So let's call getNode() method once, and save the result of this method tho the local variable